sympy/functions/combinatorial/factorials.py

```diff
@@ -15,6 +15,8 @@
 
 from sympy.polys.polytools import Poly
 
+from mpmath import gamma
```

Comment:
mpmath functions are for numeric evaluation. The argument of factorial2 could be symbolic. If you need gamma function, use SymPy's gamma function.